### PR TITLE
No bug: allow console logs for non production builds.

### PR DESCRIPTION
**Pull request by iccub**
Advanced setup taken from https://github.com/DaveWoodCom/XCGLogger#advanced-usage-recommended

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-ios/issues) for my issue if one did not already exist.
- [ ] My patch or PR title has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!` (or `No Bug: <message>` if no relevant ticket)
- [ ] *Unit Tests* are updated to cover new or changed functionality
- [ ] User-facing strings use `NSLocalizableString()`
- [ ] New files have MPL-2.0 license header.

## Reviewer Checklist:

- [ ] PR is linked to an issue via [Zenhub](https://www.zenhub.com/extension).
- [ ] Issues are assigned to at least one epic.
- [ ] Issues include necessary QA labels:
  - [ ] `QA/(Yes|No)`
  - [ ] `release-notes/(include|exclude)`
  - [ ] `bug` / `enhancement`
- [ ] Necessary security reviews have taken place.
- [ ] Adequate test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable)

### CODE REVIEW

File: Shared/RollingFileLogger.swift [hunk 1]
```diff
@@ -27,6 +27,27 @@ open class RollingFileLogger: XCGLogger {
         self.sizeLimit = sizeLimit
         self.logDirectoryPath = logDirectoryPath
         super.init()
+        
+        if !AppConstants.BuildChannel.isRelease {
```
Comment (kylehickinson) [inline Shared/RollingFileLogger.swift:33]: This is only really useful for builds made on the machine right? Is there anyway to guarantee this doesn't get called for any build on a dev machine and not to do with release configuration?

Comment (iccub) [reply Shared/RollingFileLogger.swift:33]: This should work on beta builds too, flags for release and beta are different

So it should work on dev and beta, not on release

**Comment by tmancey** [comment]
@iccub Will this redirect all console logs as ads/confirmations/ledger have important logging to help QA, thanks

**Comment by iccub** [comment]
That's correct @tmancey all Rewards logs are redirected

**Comment by kylehickinson** [comment]
Is this PR still needed?

**Pull request closed without merging**